test/test-tuntap.mjs

```diff
@@ -0,0 +1,46 @@
+import { TunTap } from "../lib/index.js";
+
+let tun;
+let shuttingDown = false;
+
+/**
+ * Cleanup function to close the TUN/TAP device.
+ * Uses a flag to ensure cleanup is only performed once.
+ */
+function cleanup() {
+  if (shuttingDown) return;
+  shuttingDown = true;
+  try {
+    // Only close if tun exists, is open, and not already closed
+    if (tun && tun.isOpen && !tun.isClosed) {
+      tun.close();
+    }
+  } catch (err) {
+    console.error("Error during cleanup:", err);
+  }
+}
+
+// Handle signals and exit
+function cleanupAndExit() {
+  cleanup();
+  process.exit(0);
+}
+
+process.on("SIGINT", cleanupAndExit);
```

Comment:
can we use try/finally instead? 
Or process.once('exit') ?

Reply:
implemented process.once('exit'..)
updated with commit https://github.com/appium/appium-ios-tuntap/commit/a02d06c5b15d4a9347470f373b856fea2273de04.